horizons/world/worldutils.py

```diff
@@ -258,18 +259,28 @@ def add_nature_objects(world, natural_resource_multiplier):
 					CreateUnit(island.worldid, UNITS.WILD_ANIMAL, x, y)(issuer=None)
 				if world.session.random.random() > WILD_ANIMAL.FOOD_AVAILABLE_ON_START:
 					building.get_component(StorageComponent).inventory.alter(RES.WILDANIMALFOOD, -1)
-			
-			if 'coastline' in tile.classes and world.session.random.random() < natural_resource_multiplier / 4.0:
-				# try to place fish: from the current position go to a random directions twice
-				for (x_dir, y_dir) in world.session.random.sample(fish_directions, 2):
-					# move a random amount in both directions
-					fish_x = x + x_dir * world.session.random.randint(3, 9)
-					fish_y = y + y_dir * world.session.random.randint(3, 9)
-					# now we have the location, check if we can build here
-					if (fish_x, fish_y) in world.ground_map:
-						Build(FishDeposit, fish_x, fish_y, world,
-						      45 + world.session.random.randint(0, 3) * 90,
-						      ownerless=True)(issuer=None)
+		
+			#list all coastline tiles
+			if 'coastline' in tile.classes:
+				coastline_tiles.append(tile)
+		
+		#island_fish_ratio is the max number of fish deposits placed by an island, len(sorted(island.ground_map.iteritems()))/50 means 1/50 coastline tiles recieve a fish deposit
+		island_fish_ratio = len(sorted(island.ground_map.iteritems()))/50
+		
+		for i in xrange(0, island_fish_ratio):
+			tile = coastline_tiles[i*(len(coastline_tiles)/island_fish_ratio)]
+			fish_deposits = 0
+			while fish_deposits == 0:
+				#randomly choose 2 directions to place the fish deposit away from the coastline
+				x_dir = fish_directions[world.session.random.randint(0, 8)][0] + fish_directions[world.session.random.randint(0, 8)][0]
```

Comment:
For the sake of readability, I'd add a shortcut for the random number generation like this:

```python
randint = world.session.random.randint
x_dir = fish_directions[randint(0, 8)][0] + fish_directions[randint(0, 8)][0]
y_dir = fish_directions[randint(0, 8)][1] + fish_directions[randint(0, 8)][1]
...
```